fdbclient/NativeAPI.actor.cpp

```diff
@@ -2383,6 +2383,39 @@ Database Database::createSimulatedExtraDatabase(std::string connectionString, Op
 	return db;
 }
 
+ACTOR Future<int64_t> fetchTenantId(DatabaseContext* cx, TenantName name) {
+	state int64_t tenantId;
+	state GetKeyServerLocationsRequest req;
+	req.tenant.name = name;
+	loop {
+		choose {
+			when(wait(cx->onProxiesChanged())) {}
+			when(GetKeyServerLocationsReply rep =
+			         wait(basicLoadBalance(cx->getCommitProxies(UseProvisionalProxies::False),
+			                               &CommitProxyInterface::getKeyServersLocations,
+			                               req,
+			                               TaskPriority::DefaultPromiseEndpoint))) {
+				tenantId = rep.tenantEntry.id;
+				break;
+			}
+		}
+	}
+	return tenantId;
+}
+
+Tenant::Tenant(DatabaseContext* cx, TenantName name) : name(name) {
+	id = fetchTenantId(cx, name);
+}
+
+ACTOR Future<int64_t> waitTenantGetIdActor(Future<int64_t> idFuture) {
+	state int64_t res = wait(idFuture);
+	return res;
+}
+
+Future<int64_t> Tenant::getId() {
+	return waitTenantGetIdActor(id);
```

Comment:
I don't think we need the actor call here, you can just return `id`:

```suggestion
	return id;
```